### PR TITLE
Move all Travis script to travis_script.py

### PR DESCRIPTION
**Pull request by hs0225**
'precommit.py' contains trvais script and building environment, which is confusing.
The next task is to rename 'precommit.py appropriately, and this should only be used to set the build environment.

IoT.js-DCO-1.0-Signed-off-by: Hosung Kim hs852.kim@samsung.com

### CODE REVIEW

**Comment by hs0225** [comment]
related issue #1229

File: .travis.yml [hunk 1]
```diff
@@ -8,38 +8,31 @@ services:
...
-  if [[ "$RUN_DOCKER" == "yes" ]]; then tools/travis_script.py;
-  else tools/precommit.py $OPTS;
-  fi
+  tools/travis_script.py
 
 env:
   global:
     - secure: "lUGzoKK/Yn4/OmpqLQALrIgfY9mQWE51deUawPrCO87UQ2GknfQ4BvwY3UT5QY0XnztPBP1+vRQ2qxbiAU7VWicp280sXDnh0FeuZD14FcE9l0FczraL12reoLu+gY5HWFfbkZncmcBsZkxDEYxhkM14FJU8fxyqGQW2ypJNz+gUGP+8r40Re5J3WjcddCQNe5IG8U+M9B4YeDHhN2QspLdN5pkgn56XtdGa3+qbecO2NpjJG5ltM9j1tTuo/Dg22DxrIFVfeFSFKUj4nfMrgPo5LevRsC/lfaBSCsj751eqrxRcQRh2hkpiIJ7mEBs2LL1EH9O6Mbj+eRh8BvIYqTB85VPNFc43sLWk14apcSVBrxJE5j3kP9sAsOD9Y5JynnkeuxYyISrkywwoX2uxsmCzIfGbwsv5VLToQzrqWlGYrHOAmVXNi8561dLfsWwxxFUjdqkZr1Kgc8UfnBEcBUtSiKCHS86/YUUbBJGkEkjDUS0GiqhFY4bXLQCR7EX4qDX3m6p7Mnh4NVUolpnSmyeYE/MjmqQ+7PJsPLL3EcIYmJ7dtW3mZ3yE2NyaFD0Pym9+TiuCCXRtrNVK1M3Kya64KNv+HbhjT/fTCgXLSeyDmJOKVAqugRlDo3b1KGR1LI0AfegzSA6mEC4e9JLjYiSnHPMUahzgLt8oU0hNFRY="
   matrix:
     - OPTS="host-linux" TARGET_OS="linux" RUN_DOCKER=yes
```
Comment (LaszloLango) [inline .travis.yml]: `OPTS` is set but not used

File: tools/travis_script.py [hunk 2]
```diff
@@ -46,44 +52,107 @@ def run_docker():
...
+                '--clean',
+                '--buildtype=' + buildtype] + args)
 
 if __name__ == '__main__':
-    run_docker()
+    if os.getenv('RUN_DOCKER') == 'yes':
+        run_docker()
 
-    test = os.environ['OPTS']
+    test = os.getenv('OPTS')
```
Comment (LaszloLango) [inline tools/travis_script.py]: Why aren't we using cmdline arguments instead? IMHO this is a bad practice.

**Comment by LaszloLango** [comment]
The title is misleading. The patch does not move the checks, but only duplicates them.

File: tools/travis_script.py (hunk 2 above)
Comment (hs0225) [reply tools/travis_script.py]: In `.travis.yml`, environment variables have been used when running scripts.

Before code,
```
`tools/precommit.py $OPTS;
```

Travis branches test with environment variables.
```
env:
   matrix:
```
If we use cmdline arguments in python script, we need to add code that handle arguments. But it also use environment variables. Do you want to put environment variables as arguments in python script as before?

Comment (LaszloLango) [reply tools/travis_script.py]: Yes, I think it is better, but it is only my personal preference.

Comment (daeyeon) [reply tools/travis_script.py]: IMHO, this script is named for travis test. So I can accept environment dependencies from travis. I assume that this script will never be locally used like how `precommit.py` was wrongly used.

File: .travis.yml (hunk 1 above)
Comment (hs0225) [reply .travis.yml]: `OPTS` is used at [this](https://github.com/hs0225/iotjs/blob/279ec1e309d34fe969ef2aa70b66868432f4f846/tools/travis_script.py#L74).

File: tools/travis_script.py (hunk 2 above)
Comment (LaszloLango) [reply tools/travis_script.py]: @daeyeon in that case we can keep it.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**